dkobo/koboform/views/survey_draft_views.py

```diff
@@ -234,16 +307,27 @@ def publish_survey_draft(request, pk, format=None):
         status_code = 504
 
     if 'formid' in resp:
```

Comment:
I have broken the long function into small functions. Checkout [this](https://github.com/onaio/dkobo/commit/d8162406d5060907be22137f466ff85cc8515b61) commit